src/sdk/comment.ts

```diff
@@ -0,0 +1,48 @@
+import { Context } from "./context";
+import { LogReturn } from "@ubiquity-os/ubiquity-os-logger";
+import { sanitizeMetadata } from "./util";
+
+const HEADER_NAME = "Ubiquity";
+
+/**
+ * Posts a comment on a GitHub issue if the issue exists in the context payload, embedding structured metadata to it.
+ */
+export async function postComment(context: Context, message: LogReturn) {
+  if ("issue" in context.payload && context.payload.repository?.owner?.login) {
+    const metadata = createStructuredMetadata(message.metadata?.name, message);
+    await context.octokit.rest.issues.createComment({
+      owner: context.payload.repository.owner.login,
+      repo: context.payload.repository.name,
+      issue_number: context.payload.issue.number,
+      body: [message.logMessage.diff, metadata].join("\n"),
+    });
+  } else {
+    context.logger.info("Cannot post comment because issue is not found in the payload");
+  }
+}
+
+function createStructuredMetadata(className: string | undefined, logReturn: LogReturn) {
+  const logMessage = logReturn.logMessage;
+  const metadata = logReturn.metadata;
+
+  const jsonPretty = sanitizeMetadata(metadata);
+  const stack = logReturn.metadata?.stack;
+  const stackLine = (Array.isArray(stack) ? stack.join("\n") : stack)?.split("\n")[2] ?? "";
+  const caller = stackLine.match(/at (\S+)/)?.[1] ?? "";
```

Comment:
The QA images I grabbed show that caller is always undefined no matter if it's `new Error` or `throw logger`

You can take `caller` from `LogReturn` it should already have it and so should `Error` too.

Reply:
All of that code was some copy paste of some function you implemented in another project I think, forgot which one, will update that.

Reply:
It's implemented in command-ask but I only added it last week and took it from this PR. It does look familiar tho I can agree with that, akin to the logger internals I thought at first

https://github.com/ubiquity-os-marketplace/command-ask/blame/eb5487d4879e48757ff758ed3fa90728469edcc1/src/handlers/comment-created-callback.ts#L63

Reply:
It's from the old [conversation rewards](https://github.com/zug-ubiquity-kernel/conversation-rewards/blame/1ebe6150d09f09229dc8b27bdbe7ec0bad2871fb/src/helpers/metadata.ts#L6) I think